tests/unit/objects/test_projects.py

```diff
@@ -142,30 +281,263 @@ def resp_list_projects():
+            url="http://localhost/api/v4/projects/2/fork/1",
             json=project_content,
             content_type="application/json",
+            status=201,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_delete_fork_relation():
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            method=responses.DELETE,
+            url="http://localhost/api/v4/projects/2/fork",
+            json=None,
+            content_type="application/json",
+            status=204,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_trigger_pipeline():
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            method=responses.POST,
+            url="http://localhost/api/v4/projects/1/trigger/pipeline",
+            json=pipeline_trigger_content,
+            content_type="application/json",
+            status=201,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_search_project_resources_by_name():
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            method=responses.GET,
+            url="http://localhost/api/v4/projects/1/search?scope=issues&search=Issue",
+            json=search_issues_content,
+            content_type="application/json",
+            status=200,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_start_housekeeping():
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            method=responses.POST,
+            url="http://localhost/api/v4/projects/1/housekeeping",
+            json="0ee4c430667fb7be8461f310",
+            content_type="application/json",
+            status=201,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_get_push_rules_project():
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            method=responses.GET,
+            url="http://localhost/api/v4/projects/1/push_rule",
+            json=push_rules_content,
+            content_type="application/json",
+            status=200,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_post_push_rules_project():
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            method=responses.POST,
+            url="http://localhost/api/v4/projects/1/push_rule",
+            json=push_rules_content,
+            content_type="application/json",
+            status=201,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_update_push_rules_project():
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            method=responses.GET,
+            url="http://localhost/api/v4/projects/1/push_rule",
+            json=push_rules_content,
+            content_type="application/json",
+            status=200,
+        )
+        rsps.add(
+            method=responses.PUT,
+            url="http://localhost/api/v4/projects/1/push_rule",
+            json=push_rules_content,
+            content_type="application/json",
+            status=201,
+        )
+        yield rsps
```

Comment:
I'm not full satisfied with this fixture because the first GET response is duplicated in [`resp_delete_push_rules_project()`](https://github.com/python-gitlab/python-gitlab/pull/2069/files#diff-e84ffbb57aae6c0765f257e13ae65e6522deac96c73b275f6450df2d9f2c5308R491)

Reply:
Thanks @antoineauger - I think what we've done in these situations is just activate responses with a decorator and explicitly add individual responses inside the test cases as needed, then assert that they were fired.

But I think it's ok to keep it like this for now - we have some other duplicates elsewhere so I think we can do a cleanup in a dedicated PR :)